### PR TITLE
JSON encodings and accompanying tests.

### LINKED ISSUE #253: JSON serialization for libsbp Haskell

### PR DESCRIPTION
**Pull request by mookerji**
/cc @mfine @kovach @fnoble

Closes #253.

### CODE REVIEW

File: haskell/src/SwiftNav/SBP/Bootload.hs [hunk 1]
```diff
@@ -11,10 +11,13 @@
 -- use only. Note that some of these messages share the same message type ID
 -- for both the host request and the device response.
 
+{-# LANGUAGE TemplateHaskell #-}
```
Comment (mfine) [inline haskell/src/SwiftNav/SBP/Bootload.hs]: could move this into cabal file

File: haskell/src/SwiftNav/SBP/Encoding.hs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+-- Module:      SwiftNav.SBP.Encoding
+-- Copyright:   Copyright (C) 2015 Swift Navigation, Inc.
+-- License:     LGPL-3
+-- Maintainer:  Mark Fine <dev@swiftnav.com>
+-- Stability:   experimental
+-- Portability: portable
+--
+-- Encoding utilities.
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
```
Comment (mfine) [inline haskell/src/SwiftNav/SBP/Encoding.hs]: can move this into the cabal file

File: haskell/src/SwiftNav/SBP/Bootload.hs [hunk 3]
```diff
@@ -68,6 +72,8 @@ instance Binary MsgBootloaderHandshakeResp where
     putWord32le msgBootloaderHandshakeRespFlags
     putByteString msgBootloaderHandshakeRespVersion
 
+$(deriveJSON defaultOptions ''MsgBootloaderHandshakeResp)
```
Comment (mfine) [inline haskell/src/SwiftNav/SBP/Bootload.hs]: out of curiosity, how does this formulate the field names - is it compatible with what we have in python? is it configurable?

File: haskell/src/SwiftNav/SBP/Bootload.hs (hunk 1 above)
Comment (mookerji) [inline haskell/src/SwiftNav/SBP/Bootload.hs]: Hmmmm, yes. Good idea.

File: haskell/src/SwiftNav/SBP/Encoding.hs (hunk 2 above)
Comment (mookerji) [inline haskell/src/SwiftNav/SBP/Encoding.hs]: Not sure. Perhaps we don't want warnings like this disabled across the entire project?

Comment (mfine) [inline haskell/src/SwiftNav/SBP/Encoding.hs]: Are they per file options? I assumed it gets turned on during the compile if this file gets brought in for everyone, so I usually just go stick it in the cabal file.

Comment (mfine) [inline haskell/src/SwiftNav/SBP/Encoding.hs]:
```NOTE: the contents of OPTIONS_GHC are prepended to the command-line options, so you do have the ability to override OPTIONS_GHC settings via the command line.

It is not recommended to move all the contents of your Makefiles into your source files, but in some circumstances, the OPTIONS_GHC pragma is the Right Thing. (If you use -keep-hc-file-too and have OPTION flags in your module, the OPTIONS_GHC will get put into the generated .hc file).```

File: haskell/src/SwiftNav/SBP/Bootload.hs (hunk 3 above)
Comment (mookerji) [inline haskell/src/SwiftNav/SBP/Bootload.hs]: Pretty sure it's just the identity (`'id`), but you can pass in some other function. See: https://hackage.haskell.org/package/aeson-0.9.0.1/docs/Data-Aeson-TH.html

**Pull request merged**